src/expr/src/scalar/func.rs

```diff
@@ -520,24 +520,16 @@ fn add_float64<'a>(a: Datum<'a>, b: Datum<'a>) -> Result<Datum<'a>, EvalError> {
     }
 }
 
-fn add_timestamp_interval<'a>(a: Datum<'a>, b: Datum<'a>) -> Result<Datum<'a>, EvalError> {
-    let dt = a.unwrap_timestamp();
-    let i = b.unwrap_interval();
-    let dt = add_timestamp_months(dt, i.months)?;
-    Ok(Datum::Timestamp(
-        dt.checked_add_signed(i.duration_as_chrono())
-            .ok_or(EvalError::TimestampOutOfRange)?,
-    ))
-}
-
-fn add_timestamptz_interval<'a>(a: Datum<'a>, b: Datum<'a>) -> Result<Datum<'a>, EvalError> {
-    let dt = a.unwrap_timestamptz().naive_utc();
-    let i = b.unwrap_interval();
-    let mut dt = add_timestamp_months(dt, i.months)?;
+fn add_timestamplike_interval<'a, T>(a: T, b: Interval) -> Result<Datum<'a>, EvalError>
+where
+    T: TimestampLike,
+{
+    let mut dt = a.date_time();
```

Comment:
Does this need to be mut? Can you use `let dt =...` multiple times below?

Reply:
Oh sure--the semantics I was trying to make clearer here is that you are performing these operations cumulatively on `dt`, but I have no strong feelings about this one way or the other.